overlord/snapstate/snapstate.go

```diff
@@ -622,6 +622,10 @@ func Install(st *state.State, name, channel string, revision snap.Revision, user
 		return nil, err
 	}
 
+	if err := snap.ValidateInstanceName(name); err != nil {
+		return nil, err
```

Comment:
Should the error here be decorated similiar to the case below ("fmt.Errorf("invalid instance name:....."))?

Reply:
It's a single snap install, so in contrast to `InstallMany()`, the error implicitly is about the current snap. That was my reasoning, but I'm happy to tweak it as needed.